Automattic/wp-calypso
# Pre-populate Add G Suite fields for first user

#### Changes proposed in this Pull Request

* Pre populates the email, first name, and last name fields of the first user since we already have something the will work for most users.

#### Testing instructions

Testing New Domain Flow:

Test Flow:
* Goto Reader, refresh page
* Goto a site
* Goto Domains
* Click "Add a domain"
* Choose any domain
* Click "Yes, add email"
* Observe that you can see the G Suite user fields filed in with your details
* Ensure add another user works
* Ensure that you can click to the next screen

Before:
![screen shot 2018-12-14 at 1 42 03 pm](https://user-images.githubusercontent.com/6817400/50026236-297d2c00-ffb6-11e8-92d3-d961c4e888dd.png)

After:
![screen shot 2018-12-14 at 3 39 54 pm](https://user-images.githubusercontent.com/6817400/50026325-8547b500-ffb6-11e8-9356-ead840b55a65.png)

Testing Existing Domain adding Email

Test Flow:
* Goto Reader, refresh page
* Goto a site with domain but no G Suite
* Goto Domains
* Goto email tab
* Click `Add G Suite` button
* Observe name is populated

Before:
![screen shot 2018-12-18 at 10 46 44 am](https://user-images.githubusercontent.com/6817400/50165195-6b59eb00-02b2-11e9-9d31-09a3295d8771.png)

After:
![screen shot 2018-12-18 at 10 46 11 am](https://user-images.githubusercontent.com/6817400/50165197-6d23ae80-02b2-11e9-9193-3edbc83ed10a.png)

Do Variants for both flows:

Variant 1
* Goto your user profile
* Change your name
* Do test flow
* Observe that name doesn't change until you refresh ( user settings are not updated in Redux state )
* Observe that name is populated correctly on refresh

Variant 2
* Goto your user profile
* Remove your name
* Refresh ( user settings are not updated in Redux state )
* Do test flow
* Observe that name is not repopulated as it doesn't exist

Note:  Names can contain invalid characters but the error handling on the form picks that up.
